### PR TITLE
Update GFE packages

### PR DESCRIPTION
**Pull request by mathomp4**
This is an omnibus update for GFE packages. The changes are:

- gFTL v1.14.0
- gFTL-shared v1.9.0
- fArgParse v1.8.0
- pFUnit v4.10.0
- yaFyaml v1.4.0

I've also integrated the changes from #44846 in what I think is a more robust way to make sure gfortran 13.3+ is not used to build yafyaml 1.3.0 and earlier.

### CODE REVIEW

**Review by tclune — APPROVED** [review]

**Comment by mathomp4** [comment]
I removed the `# generated` comments on the `depends_on()` lines per request of @haampie on slack.

**Comment by mathomp4** [comment]
I'm going to page @tldahlgren mainly to ask (from the Spack devs point of view) if there is more I need to do in this PR?

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256s, rest LGTM

File: var/spack/repos/builtin/packages/yafyaml/package.py [hunk 1]
```diff
@@ -60,6 +62,35 @@ class Yafyaml(CMakePackage):
         msg="yaFyaml only works with the Fujitsu compiler from 1.3.0 onwards",
     )
 
+    # GCC 13.3 and higher only work with yafyaml 1.4.0 onwards
+    # First we can check if the spec is gcc@13.3...
+    conflicts("%gcc@13.3:", when="@:1.3.0", msg="GCC 13.3+ only works with yafyaml 1.4.0 onwards")
+
+    # ...but if it is not (say apple-clang with gfortran as a fc), there is
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/yafyaml/package.py:71]: I am only seeing this now. There is a much simpler way to achieve the same: https://github.com/spack/spack/blob/7efbad0d812f17c2dcffe32649d9c14db82a69dd/var/spack/repos/builtin/packages/esmf/package.py#L219

Comment (mathomp4) [reply var/spack/repos/builtin/packages/yafyaml/package.py:71]: Ahh. I didn't know about `compiler_environment()`. I'll take a look at updating the code...

**Pull request merged**